keps/sig-node/3619-supplemental-groups-policy/README.md

```diff
@@ -884,6 +937,15 @@ Describe the metrics themselves and the reasons why they weren't added (e.g., co
 implementation difficulties, etc.).
 -->
 
+Potentially, kube-scheduler could be implemented to avoid scheduling a pod with `supplementalGroupsPolicy: Strict`
+to a node running CRI runtime which does not supported this feature.
+
+In this way, the Event metric described above would not happen, and users would instead see `Pending` pods
+as an error metric.
+
+However, this is not planned to be implemented in kube-scheduler, as it seems overengineering.
+Users may use `nodeSelector`, `nodeAffinity`, etc. to workaround this.
```

Comment:
Ok, this needs to be clearly documented, I didn't see this anywhere above (I may have missed it). Is the assumption then that during the time period while this is working its way through the ecosystem (maybe a couple years?), NFD/a label + node selector will be needed to ensure that workloads that need the `Strict` policy are properly scheduled?

Reply:
> NFD/a label + node selector will be needed to ensure that workloads that need the Strict policy are properly scheduled?

Yes. However, this topic is not specific to the KEP. This is common issue for node/runtime handler features (e.g. user namespaces, recursive read-only mounts.)

Reply:
yes...and I think this is a big usability oversight. but you are right, that's not specific to this KEP.

Reply:
I agree it's miserable and the containerd 2.0 situation is exaccerbating it.

The only paths I see are basically:

1) Back-rev nodes ignore unknown fields; the API may say one thing and the node does something else.

2) Scheduler does not know about features, and so might assign pods which use feature X to nodes which do not support feature X; back-rev nodes will detect the "unknown fields" and reject those pods.

3) Scheduler knows how to map pods which need feature X to nodes which support feature X; scheduling may fail entirely if no nodes support it.

4) We stall any feature which has a node-based implementation until old kubelets are out of support; CRI support is not version locked or controlled by us, so fall back on 1, 2, or 3.